extensions/amp-analytics-resources/0.1/analytics-resources-impl.js

```diff
@@ -0,0 +1,192 @@
+/**
+ * Copyright 2016 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {getService} from '../../../src/service';
+import {parseUrl} from '../../../src/url';
+import {
+  getResourceTiming as rtcGetResourceTiming,
+} from 'resourcetiming-compression/src/resourcetiming-compression';
+
+/** @private Visible for testing. */
+export class AnalyticsResourcesService {
+  constructor(win) {
+    /** @private @const {!Window} */
+    this.win_ = win;
+  }
+
+  /**
+   * Returns rendered byte count of base page. This may not be what was sent on the network.
+   * @return {!Promise<string|undefined>}
+   */
+  getDocumentLength() {
+    const innerHTML = this.win_['document'] &&
+        this.win_['document']['documentElement'] &&
+        this.win_['document']['documentElement']['innerHTML'];
+    if (!innerHTML || typeof innerHTML.length !== 'number') {
+      return Promise.resolve();
+    }
+    return Promise.resolve(String(innerHTML.length));
+  }
+
+  /**
+   * Returns number of resources in ResourceTiming.
+   * @return {!Promise<string|undefined>}
+   */
+  getResourceCount() {
+    const res = this.getResources_();
+    return (res && typeof res.length === 'number') ?
+        Promise.resolve(String(res.length)) : Promise.resolve();
+  }
+
+  /**
+   * Returns number of DOM nodes on the page.
+   * @return {!Promise<string|undefined>}
+   */
+  getDomNodeCount() {
+    const nodes = this.getNodes_('*');
+    return (nodes && typeof nodes.length === 'number') ?
+        Promise.resolve(String(nodes.length)) : Promise.resolve();
+  }
+
+  /**
+   * Returns number of image nodes on the page.
+   * @return {!Promise<string|undefined>}
+   */
+  getDomImgCount() {
+    const nodes = this.getNodes_('img');
```

Comment:
I'm wondering if this is what you actually want. In AMP `img` tags are created lazily as they are the user scrolls through the content. Thus, some `amp-img` might exist but not yet "loaded" their respective `img` tags.